src/glyph/builder.rs

```diff
@@ -316,125 +356,100 @@ impl GlyphBuilder {
 
     /// Ends the path begun by `begin_path()` and adds the contour it to the glyph's outline.
     ///
-    /// Errors when:
-    /// 1. `outline()` wasn't called first.
-    /// 2. `begin_path()` wasn't called first.
-    /// 3. the point sequence is forbidden by the specification.
-    ///
     /// Discards path in case of error.
+    ///
+    /// Errors when:
+    /// 1. `begin_path()` wasn't called first.
+    /// 2. the point sequence is forbidden by the specification.
     pub fn end_path(&mut self) -> Result<&mut Self, ErrorKind> {
-        if self.glyph.outline.is_none() {
-            return Err(ErrorKind::UnexpectedDrawing);
-        }
-
         let contour = self.scratch_contour.take();
         match contour {
-            Some(mut c) => {
-                // If using format version 1, check if we are actually looking at an implicit
-                // anchor and convert if so (leaving an empty outline if there is nothing else).
-                if Self::contour_is_v1_anchor(self.get_format(), &c.points) {
-                    let anchor_point = c.points.remove(0);
-                    let anchor = Anchor {
-                        name: anchor_point.name,
-                        x: anchor_point.x,
-                        y: anchor_point.y,
-                        identifier: None,
-                        color: None,
-                    };
-                    self.glyph.anchors.get_or_insert(Vec::new()).push(anchor);
-                    Ok(self)
+            Some(c) => {
                 // If ending a closed contour with off-curve points, wrap around and check
                 // from the beginning that we have a curve or qcurve following eventually.
-                } else {
-                    if self.number_of_offcurves > 0 {
-                        if c.is_closed() {
-                            for point in &c.points {
-                                match point.typ {
-                                    PointType::OffCurve => self.number_of_offcurves += 1,
-                                    PointType::QCurve => break,
-                                    PointType::Curve => {
-                                        if self.number_of_offcurves > 2 {
-                                            return Err(ErrorKind::TooManyOffCurves);
-                                        }
-                                        break;
-                                    }
-                                    PointType::Line => {
-                                        return Err(ErrorKind::UnexpectedPointAfterOffCurve);
+                if self.number_of_offcurves > 0 {
+                    if c.is_closed() {
+                        for point in &c.points {
+                            match point.typ {
+                                PointType::OffCurve => {
+                                    self.number_of_offcurves =
+                                        self.number_of_offcurves.saturating_add(1)
```

Comment:
so diligent!

realistically, it seems unlikely that we have more that `u32::MAX` preceding off-curve points, but honestly I think using the checked arithmetic methods is a good habit. 👍

Reply:
Can't stand edge cases!